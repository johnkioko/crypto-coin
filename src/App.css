@import url("https://fonts.googleapis.com/css2?family=Montserrat:ital,wght@0,100;0,200;0,300;0,400;0,500;1,100;1,200;1,300;1,400&family=Roboto+Slab:wght@100;200;300;400&display=swap");
@import url("https://fonts.googleapis.com/css2?family=Raleway:wght@100;300;400;500;600&display=swap");
@import url("https://fonts.googleapis.com/css2?family=Nunito+Sans:wght@200;400;600&display=swap");

* {
  box-sizing: border-box;
  padding: 0;
  margin: 0;
  font-family: "Montserrat", sans-serif;
}

::-webkit-scrollbar {
  width: 10px;
}

::-webkit-scrollbar-track {
  box-shadow: 0 10px 10px 5px rgb(255, 255, 255);
  border-radius: 10px;
}

::-webkit-scrollbar-thumb {
  background: var(--color2);
  border-radius: 10px;
  border: 1px solid rgb(255, 255, 255, 0.4);
}

:root {
  --color1: #4369b1;
  --color2: #5687e3;
}

.white {
  color: #fff;
}

.flex {
  display: flex;
  align-items: center;
}

.p-one {
  padding: 5px;
}

.p-two {
  padding: 20px;
}

.left {
  text-align: left;
}

.h-one {
  font-weight: bold;
}

.font-2 {
  font-family: "Roboto Slab", serif;
}

.font-3 {
  font-family: "Raleway", sans-serif;
  font-weight: 300;
}

.font-4 {
  font-family: "Nunito Sans", sans-serif;
  font-weight: 600;
}

body {
  background-color: var(--color1);
}

.search-field {
  width: 100%;
  outline: 0;
  padding: 5px;
  border: none;
  border-radius: 5px;
}

.header {
  width: 100%;
  justify-content: space-between;
  border: 1px solid var(--color1);
  border-radius: 5px;
  margin: auto;
  border-bottom: 1px solid rgb(255, 255, 255, 0.5);
}

.header-detail {
  justify-content: space-between;
  width: 100%;
  border-bottom: 1px solid var(--color1);
  padding: 10px;
}

.logo-header-detail {
  width: 70%;
  justify-content: space-between;
}

.fa {
  color: #fff;
  cursor: pointer;
  font-size: 50px;
}

.right-side {
  gap: 10px;
  padding: 10px;
  justify-content: flex-end;
}

.logo-header {
  padding: 10px;
  gap: 5px;
  width: 100%;
  justify-content: space-between;
}

.logo {
  width: 50%;
  border-radius: 10px;
}

.fa-chevron-left {
  padding-right: 10px;
}

.home-heading {
  align-items: center;
}

.wrapper {
  background: var(--color1);
  flex-basis: 12em;
  width: 100%;
  height: 10em;
}

.wrapper2 {
  background: var(--color1);
  flex-basis: 12em;
  width: 100%;
  height: 10em;
}

.bg-wrapper {
  background: url("./img/diversification.png") no-repeat center;
  background-size: contain;
  flex-basis: 12em;
  width: 100%;
  border-radius: 6px;
  height: 100px;
  opacity: 0.5;
  flex-grow: 1;
}

.bg-wrapper2 {
  background: url("./img/cryptocurrency.png") no-repeat center;
  background-size: contain;
  flex-basis: 12em;
  width: 100%;
  border-radius: 6px;
  height: 100px;
  opacity: 0.5;
}

.wrapper-content {
  width: 50%;
  background: #5687e357;
  height: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  gap: 10px;
}

.wrapper-content-2 {
  width: 50%;
  background: #5687e357;
  height: 100%;
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  text-align: right;
}

.crypto-details {
  flex-wrap: wrap;
  width: 100%;
  margin: auto;
  flex: 1;
}

.crypto-details > button:nth-child(4n + 0),
.crypto-details > button:nth-child(4n + 1) {
  background: #4167ae;
}

.crypto-details > button:nth-child(4n + 2),
.crypto-details > button:nth-child(4n + 3) {
  background: #4d76c2;
}

.details-container {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: flex-end;
  width: 50%;
  outline: 0;
  border: 0;
  height: 170px;
  cursor: pointer;
}

tr > td > p {
  width: 100%;
  margin: 0 auto;
  padding: 10px 20px 10px 20px;
  background: #00be3f;
  border-radius: 5px;
  border: 0;
  color: #fff;
  font-weight: bold;
  font-size: 15px;
}

tr > td > p > a {
  text-decoration: none;
  color: #fff;
}

tr {
  width: 100%;
  display: flex;
  justify-content: space-between;
}

td {
  padding: 20px;
}

tr > td:nth-child(1) {
  font-weight: bold;
  font-size: 15px;
}

tr:nth-child(odd) {
  background: #4c73bb;
}

tr:nth-child(even) {
  background: #3f62a4;
}

.crypto-name {
  display: flex;
  flex-direction: column;
  gap: 10px;
}

.crypto-name > p {
  text-align: right;
}

.crypto-info {
  font-weight: bold;
  font-size: 20px;
}

.price,
.change {
  font-size: 15px;
  font-weight: normal;
}

.symbol {
  padding-left: 5px;
}

table {
  width: 100%;
  border-collapse: collapse;
  border-spacing: 0;
}

.rank {
  font-family: "Roboto Slab", serif;
}

.status {
  padding: 5px;
}

.fa-caret-up {
  color: rgb(0, 255, 0);
  padding-left: 5px;
}

.fa-caret-down {
  color: red;
  padding-left: 5px;
}

.load-spinner {
  height: 50vh;
  width: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  position: absolute;
}

.search {
  width: 100%;
  display: flex;
  justify-content: center;
  padding: 10px;
  margin-top: 2rem;
  border-radius: 1rem;

  /* border-bottom: 1px solid rgb(0, 255, 0, 0.5);
  border-top: 1px solid rgb(0, 255, 0, 0.5); */
}

.search > input {
  padding: 10px;
  border-radius: 0;
  border: none;
  outline: none;
  width: 60%;
  margin: auto;
}

.search > input::placeholder {
  color: var(--color2);
}

.search-msg {
  display: none;
  padding: 10px;
  width: 80%;
  margin: auto;
  border-radius: 5px;
  text-align: center;
}

.failure {
  background: rgb(255, 0, 0);
  color: rgb(255, 255, 255);
}

.success {
  background: #00be3f;
  color: #fff;
  font-weight: 400;
}

@media screen and (min-width: 768px) {
  .header,
  .header-detail {
    width: 100%;
    padding: 20px;
  }

  .logo-header,
  .right-side,
  .logo-header-detail {
    flex-grow: 1;
  }

  .wrapper {
    width: 100%;
  }

  .bg-wrapper,
  .bg-wrapper2 {
    flex-grow: 1;
  }

  .tr > td > p:hover {
    background: #004216;
  }
}
