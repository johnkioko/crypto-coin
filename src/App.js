import { Route, Routes } from 'react-router-dom';
import './App.css';
import Crytpos from './Components/Crytpos';
import Crypto from './Components/Crypto';
import NavBar from './Components/NavBar';
import DetailsPage from './Components/DetailsPage';

function App() {
  return (
    <Routes>
      <Route
        path="/"
        element={(
          <>
            <NavBar />
            <Crytpos />
          </>
        )}
      />
      <Route
        path="/crypto-details"
        element={(
          <>
            <DetailsPage />
            <Crypto />
          </>
        )}
      />
    </Routes>
  );
}

export default App;
