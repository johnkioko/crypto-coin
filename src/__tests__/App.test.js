import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import App from '../App';
import Store from '../Redux/ConfigureStore';

const MockApp = () => (
  <BrowserRouter>
    <Provider store={Store}>
      <App />
    </Provider>
  </BrowserRouter>
);

describe('App', () => {
  render(<MockApp />);

  it('renders without crashing', async () => {
    expect(render(<MockApp />)).toMatchSnapshot();
    expect(true).toBeTruthy();
  });
});
