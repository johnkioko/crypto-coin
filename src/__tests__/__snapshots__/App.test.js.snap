// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without crashing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="header flex"
      >
        <div
          class="logo-header flex"
        >
          <div
            class="logo flex"
          >
            <span
              class="back"
            >
              <span
                class="back-txt white"
              >
                <h1
                  class="logo"
                >
                  CRYPTO
                </h1>
              </span>
            </span>
          </div>
        </div>
        <div
          class="right-side flex"
        >
          <span
            class="audio"
          >
            <i
              aria-hidden="true"
              class="fa fa-microphone"
            />
          </span>
          <span
            class="settings"
          >
            <i
              aria-hidden="true"
              class="fa fa-cog"
            />
          </span>
        </div>
      </header>
      <div
        class="cryptos-container"
      >
        <div
          class="wrapper flex"
        >
          <div
            class="bg-wrapper"
          />
          <div
            class="wrapper-content p-two"
          >
            <h4
              class="white left h-one"
            >
              CYRPTO CURRENCIES
            </h4>
            <p
              class="white left"
            >
               
              Cyrptos
            </p>
          </div>
        </div>
        <div
          class="search"
        >
          <input
            name="search"
            placeholder="Search..."
            type="text"
            value=""
          />
        </div>
        <h4
          class="white p-two"
        >
          CRYPTO CURRENCY STATS
        </h4>
        <p
          class="search-msg"
          style="display: none;"
        >
          No Cyrptos Found.
        </p>
        <div
          class="crypto-details flex"
        >
          <div>
            <div
              class="load-spinner"
            >
              <span
                size="500"
                style="display: inherit; position: relative; font-size: 0px; top: 20px; left: 20px; width: 60px; height: 60px;"
              >
                <span
                  style="position: absolute; width: 5px; height: 15px; margin: 2px; background-color: rgb(54, 215, 183); border-radius: 2px; transition: 2s; animation-fill-mode: both; animation: react-spinners-FadeLoader-fade 1.2s 0.12s infinite ease-in-out; top: 20px; left: 0px;"
                />
                <span
                  style="position: absolute; width: 5px; height: 15px; margin: 2px; background-color: rgb(54, 215, 183); border-radius: 2px; transition: 2s; animation-fill-mode: both; animation: react-spinners-FadeLoader-fade 1.2s 0.24s infinite ease-in-out; top: 13.636363636363637px; left: 13.636363636363637px; transform: rotate(-45deg);"
                />
                <span
                  style="position: absolute; width: 5px; height: 15px; margin: 2px; background-color: rgb(54, 215, 183); border-radius: 2px; transition: 2s; animation-fill-mode: both; animation: react-spinners-FadeLoader-fade 1.2s 0.36s infinite ease-in-out; top: 0px; left: 20px; transform: rotate(90deg);"
                />
                <span
                  style="position: absolute; width: 5px; height: 15px; margin: 2px; background-color: rgb(54, 215, 183); border-radius: 2px; transition: 2s; animation-fill-mode: both; animation: react-spinners-FadeLoader-fade 1.2s 0.48s infinite ease-in-out; top: -13.636363636363637px; left: 13.636363636363637px; transform: rotate(45deg);"
                />
                <span
                  style="position: absolute; width: 5px; height: 15px; margin: 2px; background-color: rgb(54, 215, 183); border-radius: 2px; transition: 2s; animation-fill-mode: both; animation: react-spinners-FadeLoader-fade 1.2s 0.6s infinite ease-in-out; top: -20px; left: 0px;"
                />
                <span
                  style="position: absolute; width: 5px; height: 15px; margin: 2px; background-color: rgb(54, 215, 183); border-radius: 2px; transition: 2s; animation-fill-mode: both; animation: react-spinners-FadeLoader-fade 1.2s 0.72s infinite ease-in-out; top: -13.636363636363637px; left: -13.636363636363637px; transform: rotate(-45deg);"
                />
                <span
                  style="position: absolute; width: 5px; height: 15px; margin: 2px; background-color: rgb(54, 215, 183); border-radius: 2px; transition: 2s; animation-fill-mode: both; animation: react-spinners-FadeLoader-fade 1.2s 0.84s infinite ease-in-out; top: 0px; left: -20px; transform: rotate(90deg);"
                />
                <span
                  style="position: absolute; width: 5px; height: 15px; margin: 2px; background-color: rgb(54, 215, 183); border-radius: 2px; transition: 2s; animation-fill-mode: both; animation: react-spinners-FadeLoader-fade 1.2s 0.96s infinite ease-in-out; top: 13.636363636363637px; left: -13.636363636363637px; transform: rotate(45deg);"
                />
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div>
      <header
        class="header flex"
      >
        <div
          class="logo-header flex"
        >
          <div
            class="logo flex"
          >
            <span
              class="back"
            >
              <span
                class="back-txt white"
              >
                <h1
                  class="logo"
                >
                  CRYPTO
                </h1>
              </span>
            </span>
          </div>
        </div>
        <div
          class="right-side flex"
        >
          <span
            class="audio"
          >
            <i
              aria-hidden="true"
              class="fa fa-microphone"
            />
          </span>
          <span
            class="settings"
          >
            <i
              aria-hidden="true"
              class="fa fa-cog"
            />
          </span>
        </div>
      </header>
      <div
        class="cryptos-container"
      >
        <div
          class="wrapper flex"
        >
          <div
            class="bg-wrapper"
          />
          <div
            class="wrapper-content p-two"
          >
            <h4
              class="white left h-one"
            >
              CYRPTO CURRENCIES
            </h4>
            <p
              class="white left"
            >
               
              Cyrptos
            </p>
          </div>
        </div>
        <div
          class="search"
        >
          <input
            name="search"
            placeholder="Search..."
            type="text"
            value=""
          />
        </div>
        <h4
          class="white p-two"
        >
          CRYPTO CURRENCY STATS
        </h4>
        <p
          class="search-msg"
        >
          No Cyrptos Found.
        </p>
        <div
          class="crypto-details flex"
        >
          <div>
            <div
              class="load-spinner"
            >
              <span
                size="500"
                style="display: inherit; position: relative; font-size: 0px; top: 20px; left: 20px; width: 60px; height: 60px;"
              >
                <span
                  style="position: absolute; width: 5px; height: 15px; margin: 2px; background-color: rgb(54, 215, 183); border-radius: 2px; transition: 2s; animation-fill-mode: both; animation: react-spinners-FadeLoader-fade 1.2s 0.12s infinite ease-in-out; top: 20px; left: 0px;"
                />
                <span
                  style="position: absolute; width: 5px; height: 15px; margin: 2px; background-color: rgb(54, 215, 183); border-radius: 2px; transition: 2s; animation-fill-mode: both; animation: react-spinners-FadeLoader-fade 1.2s 0.24s infinite ease-in-out; top: 13.636363636363637px; left: 13.636363636363637px; transform: rotate(-45deg);"
                />
                <span
                  style="position: absolute; width: 5px; height: 15px; margin: 2px; background-color: rgb(54, 215, 183); border-radius: 2px; transition: 2s; animation-fill-mode: both; animation: react-spinners-FadeLoader-fade 1.2s 0.36s infinite ease-in-out; top: 0px; left: 20px; transform: rotate(90deg);"
                />
                <span
                  style="position: absolute; width: 5px; height: 15px; margin: 2px; background-color: rgb(54, 215, 183); border-radius: 2px; transition: 2s; animation-fill-mode: both; animation: react-spinners-FadeLoader-fade 1.2s 0.48s infinite ease-in-out; top: -13.636363636363637px; left: 13.636363636363637px; transform: rotate(45deg);"
                />
                <span
                  style="position: absolute; width: 5px; height: 15px; margin: 2px; background-color: rgb(54, 215, 183); border-radius: 2px; transition: 2s; animation-fill-mode: both; animation: react-spinners-FadeLoader-fade 1.2s 0.6s infinite ease-in-out; top: -20px; left: 0px;"
                />
                <span
                  style="position: absolute; width: 5px; height: 15px; margin: 2px; background-color: rgb(54, 215, 183); border-radius: 2px; transition: 2s; animation-fill-mode: both; animation: react-spinners-FadeLoader-fade 1.2s 0.72s infinite ease-in-out; top: -13.636363636363637px; left: -13.636363636363637px; transform: rotate(-45deg);"
                />
                <span
                  style="position: absolute; width: 5px; height: 15px; margin: 2px; background-color: rgb(54, 215, 183); border-radius: 2px; transition: 2s; animation-fill-mode: both; animation: react-spinners-FadeLoader-fade 1.2s 0.84s infinite ease-in-out; top: 0px; left: -20px; transform: rotate(90deg);"
                />
                <span
                  style="position: absolute; width: 5px; height: 15px; margin: 2px; background-color: rgb(54, 215, 183); border-radius: 2px; transition: 2s; animation-fill-mode: both; animation: react-spinners-FadeLoader-fade 1.2s 0.96s infinite ease-in-out; top: 13.636363636363637px; left: -13.636363636363637px; transform: rotate(45deg);"
                />
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <header
      class="header flex"
    >
      <div
        class="logo-header flex"
      >
        <div
          class="logo flex"
        >
          <span
            class="back"
          >
            <span
              class="back-txt white"
            >
              <h1
                class="logo"
              >
                CRYPTO
              </h1>
            </span>
          </span>
        </div>
      </div>
      <div
        class="right-side flex"
      >
        <span
          class="audio"
        >
          <i
            aria-hidden="true"
            class="fa fa-microphone"
          />
        </span>
        <span
          class="settings"
        >
          <i
            aria-hidden="true"
            class="fa fa-cog"
          />
        </span>
      </div>
    </header>
    <div
      class="cryptos-container"
    >
      <div
        class="wrapper flex"
      >
        <div
          class="bg-wrapper"
        />
        <div
          class="wrapper-content p-two"
        >
          <h4
            class="white left h-one"
          >
            CYRPTO CURRENCIES
          </h4>
          <p
            class="white left"
          >
             
            Cyrptos
          </p>
        </div>
      </div>
      <div
        class="search"
      >
        <input
          name="search"
          placeholder="Search..."
          type="text"
          value=""
        />
      </div>
      <h4
        class="white p-two"
      >
        CRYPTO CURRENCY STATS
      </h4>
      <p
        class="search-msg"
      >
        No Cyrptos Found.
      </p>
      <div
        class="crypto-details flex"
      >
        <div>
          <div
            class="load-spinner"
          >
            <span
              size="500"
              style="display: inherit; position: relative; font-size: 0px; top: 20px; left: 20px; width: 60px; height: 60px;"
            >
              <span
                style="position: absolute; width: 5px; height: 15px; margin: 2px; background-color: rgb(54, 215, 183); border-radius: 2px; transition: 2s; animation-fill-mode: both; animation: react-spinners-FadeLoader-fade 1.2s 0.12s infinite ease-in-out; top: 20px; left: 0px;"
              />
              <span
                style="position: absolute; width: 5px; height: 15px; margin: 2px; background-color: rgb(54, 215, 183); border-radius: 2px; transition: 2s; animation-fill-mode: both; animation: react-spinners-FadeLoader-fade 1.2s 0.24s infinite ease-in-out; top: 13.636363636363637px; left: 13.636363636363637px; transform: rotate(-45deg);"
              />
              <span
                style="position: absolute; width: 5px; height: 15px; margin: 2px; background-color: rgb(54, 215, 183); border-radius: 2px; transition: 2s; animation-fill-mode: both; animation: react-spinners-FadeLoader-fade 1.2s 0.36s infinite ease-in-out; top: 0px; left: 20px; transform: rotate(90deg);"
              />
              <span
                style="position: absolute; width: 5px; height: 15px; margin: 2px; background-color: rgb(54, 215, 183); border-radius: 2px; transition: 2s; animation-fill-mode: both; animation: react-spinners-FadeLoader-fade 1.2s 0.48s infinite ease-in-out; top: -13.636363636363637px; left: 13.636363636363637px; transform: rotate(45deg);"
              />
              <span
                style="position: absolute; width: 5px; height: 15px; margin: 2px; background-color: rgb(54, 215, 183); border-radius: 2px; transition: 2s; animation-fill-mode: both; animation: react-spinners-FadeLoader-fade 1.2s 0.6s infinite ease-in-out; top: -20px; left: 0px;"
              />
              <span
                style="position: absolute; width: 5px; height: 15px; margin: 2px; background-color: rgb(54, 215, 183); border-radius: 2px; transition: 2s; animation-fill-mode: both; animation: react-spinners-FadeLoader-fade 1.2s 0.72s infinite ease-in-out; top: -13.636363636363637px; left: -13.636363636363637px; transform: rotate(-45deg);"
              />
              <span
                style="position: absolute; width: 5px; height: 15px; margin: 2px; background-color: rgb(54, 215, 183); border-radius: 2px; transition: 2s; animation-fill-mode: both; animation: react-spinners-FadeLoader-fade 1.2s 0.84s infinite ease-in-out; top: 0px; left: -20px; transform: rotate(90deg);"
              />
              <span
                style="position: absolute; width: 5px; height: 15px; margin: 2px; background-color: rgb(54, 215, 183); border-radius: 2px; transition: 2s; animation-fill-mode: both; animation: react-spinners-FadeLoader-fade 1.2s 0.96s infinite ease-in-out; top: 13.636363636363637px; left: -13.636363636363637px; transform: rotate(45deg);"
              />
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
